exchange/c2cx/integration_test.go

```diff
@@ -1,18 +1,898 @@
+	Asks      []MarketOrderString
+}
+
+// Orderbook with timestamp
+type orderbookTest struct {
+	TradePair TradePair
+	Timestamp time.Time
+	Bids      exchange.MarketOrders
+	Asks      exchange.MarketOrders
+}
+
+// UnmarshalJSON implements json.Unmarshaler
+func (r *orderbookTest) UnmarshalJSON(b []byte) error {
+	var v orderbookJSONTestResponse
+	err := json.Unmarshal(b, &v)
+	if err != nil {
+		return err
+	}
+
+	t, err := time.Parse(time.RFC3339, *v.Timestamp)
+
+	if err != nil {
+		return err
+	}
+	r.Timestamp = t
+
+	r.Bids = make(exchange.MarketOrders, len(v.Bids))
+	for i := 0; i < len(v.Bids); i++ {
+		price, err := decimal.NewFromString(v.Bids[i].Price)
+		if err != nil {
+			return err
+		}
+		volume, err := decimal.NewFromString(v.Bids[i].Volume)
+		if err != nil {
+			return err
+		}
+		r.Bids[i] = exchange.MarketOrder{
+			Price:  price,
+			Volume: volume,
 		}
-		panic("C2CX secret not provided")
 	}
-	panic("C2CX key not provided")
-}()
+
+	return nil
+}
+
+func TestGetOrderbook(t *testing.T) {
+	tt := []struct {
+		name       string
+		args       []string
+		errMessage string
+	}{
+		{
+			name:       "get_orderbook - symbol not exists",
+			args:       []string{"get_orderbook", "USDT_BT"},
+			errMessage: "C2CX request failed: endpoint=getorderbook code=400 message=symbol not exists\n",
+		},
+		{
+			name: "get_orderbook - OK",
+			args: []string{"get_orderbook", "USDT_BTG"},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+
+			output, err := exec.Command(binaryPath, tc.args...).CombinedOutput()
+			if tc.errMessage != "" && err != nil {
+				require.Equal(t, tc.errMessage, string(output))
+				return
+			}
+			require.NoError(t, err)
+			var resp orderbookTest
+			err = resp.UnmarshalJSON(output)
+			require.NoError(t, err)
+			for _, bid := range resp.Bids {
+				require.True(t, bid.Volume.Cmp(decimal.New(0, 10)) > 0)
+				require.True(t, bid.Price.Cmp(decimal.New(0, 10)) > 0)
+			}
+
+			for _, ask := range resp.Asks {
+				require.True(t, ask.Volume.Cmp(decimal.New(0, 10)) > 0)
+				require.True(t, ask.Price.Cmp(decimal.New(0, 10)) > 0)
+			}
+
+		})
+	}
+}
+
+// BalanceSummaryString includes the account balance and its frozen balance
+type BalanceSummaryString struct {
+	Balance map[string]string `json:"balance"`
+	Frozen  map[string]string `json:"frozen"`
+}
+
+func TestClient_GetBalanceSummary(t *testing.T) {
+	tt := []struct {
+		name       string
+		args       []string
+		errMessage string
+	}{
+		{
+			name: "get_balance_summary",
+			args: []string{"get_balance_summary"},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			output, err := exec.Command(binaryPath, tc.args...).CombinedOutput()
+			require.NoError(t, err)
+			fmt.Println(string(output))
+			var resp BalanceSummaryString
+			err = json.Unmarshal(output, &resp)
+			require.NoError(t, err)
+			require.True(t, len(resp.Balance) > 0)
+			require.True(t, len(resp.Frozen) > 0)
```

Comment:
can you check that the final calculated balance (which subtracts frozen) succeeds and have valid values (`>=0`)?